python/adjoint/objective.py

```diff
@@ -99,6 +99,13 @@ def _adj_src_scale(self, include_resolution=True):
             scale *= 2
         return scale
 
+    def _dft_monitor_size(self,min_max_corners):
+        """Evaluates the size of the dft monitor for FourierFields."""
+        thickness_indicator = self.sim.fields.indicate_thick_dims(self.volume.swigobj)
+        raw_size = (min_max_corners[3:6]-min_max_corners[0:3])/2+np.ones(3)
+        true_size = np.array([raw_size[i]*(thickness_indicator[i]==1)+(thickness_indicator[i]==0) for i in range(3)],dtype=int)
+        return true_size
```

Comment:
You should really be using the *same* code that is used to construct the DFT slice to begin with, not reconstructing the slice dimensions.   

This means refactoring this code:  https://github.com/NanoComp/meep/blob/1c7a1edd86a378c8dcd3dcd3e2e1ed40620fe956/src/dft.cpp#L1044-L1067

to pull out the computation of the DFT slice dimensions into a separate function, so that you can call it in *both* `process_dft_component` and in your code.

Reply:
also this code:
https://github.com/NanoComp/meep/blob/1c7a1edd86a378c8dcd3dcd3e2e1ed40620fe956/src/dft.cpp#L1069-L1096

Reply:
> to pull out the computation of the DFT slice dimensions into a separate function, so that you can call it in both process_dft_component and in your code.

Like `process_dft_component`, this _separate function_ would be a member function of `fields`, but the function `fourier_sourcedata` in my code, which needs the output of this _separate function_, is a member function of `dft_fields`. Would it be acceptable in Meep if a `fields` function is made callable from `dft_fields`?